include/nbl/video/IDeviceMemoryAllocation.h

```diff
@@ -24,6 +24,8 @@ We only support persistently mapped buffers with ARB_buffer_storage.
 Please don't ask us to support Buffer Orphaning. */
 class IDeviceMemoryAllocation : public virtual core::IReferenceCounted
 {
+    friend class IDeviceMemoryAllocator;
+    friend class ILogicalDevice;
```

Comment:
gimme a reason

Reply:
needed [here](https://github.com/Devsh-Graphics-Programming/Nabla/blob/bd32f3617632811af2028abf273bf59bcdb6d060/include/nbl/video/IDeviceMemoryAllocator.h#L126)

Reply:
you definitely don't need a friendship with `ILogicalDevice` because then whatever other thing might inherit from`IDeviceMemoryAllocator` would need the friendship declared here too.